### ISSUE TITLE
UnityTimeOutException: The Unity environment took too long to respond.

### ISSUE THREAD

**Issue by RaStall** [Unity-Technologies/ml-agents#3683]
**Describe the bug**
I cannot run training in Unity editor. When Issuing command `mlagents-learn config/trainer_config.yaml --train` I get

> mlagents_envs.exception.UnityTimeOutException: The Unity environment took too long to respond. Make sure that :
         The environment does not need user interaction to launch
         The Agents are linked to the appropriate Brains
         The environment and the Python interface have compatible versions.

When running via same command but by pointing to build .exe file(`mlagents-learn --env=../foo.exe config/trainer_config.yaml --train`), all works fine.

**To Reproduce**
1. Run Unity and open Project using ml-agents
2. Open windows command line and issue command `mlagents-learn config/trainer_config.yaml --train`

**Console logs / stack traces**
```C:\Users\korfanty\Unity\ml-agents>mlagents-learn config/trainer_config.yaml --curriculum config/curricula/penguin/PenguinLearning.yaml --run-id=Penguin1 --train
WARNING:tensorflow:From c:\users\korfanty\unity\python3.7\lib\site-packages\tensorflow_core\python\compat\v2_compat.py:65: disable_resource_variables (from tensorflow.python.ops.variable_scope) is deprecated and will be removed in a future version.
Instructions for updating:
non-resource variables are not supported in the long term


                        ▄▄▄▓▓▓▓
                   ╓▓▓▓▓▓▓█▓▓▓▓▓
              ,▄▄▄m▀▀▀'  ,▓▓▓▀▓▓▄                           ▓▓▓  ▓▓▌
            ▄▓▓▓▀'      ▄▓▓▀  ▓▓▓      ▄▄     ▄▄ ,▄▄ ▄▄▄▄   ,▄▄ ▄▓▓▌▄ ▄▄▄    ,▄▄
          ▄▓▓▓▀        ▄▓▓▀   ▐▓▓▌     ▓▓▌   ▐▓▓ ▐▓▓▓▀▀▀▓▓▌ ▓▓▓ ▀▓▓▌▀ ^▓▓▌  ╒▓▓▌
        ▄▓▓▓▓▓▄▄▄▄▄▄▄▄▓▓▓      ▓▀      ▓▓▌   ▐▓▓ ▐▓▓    ▓▓▓ ▓▓▓  ▓▓▌   ▐▓▓▄ ▓▓▌
        ▀▓▓▓▓▀▀▀▀▀▀▀▀▀▀▓▓▄     ▓▓      ▓▓▌   ▐▓▓ ▐▓▓    ▓▓▓ ▓▓▓  ▓▓▌    ▐▓▓▐▓▓
          ^█▓▓▓        ▀▓▓▄   ▐▓▓▌     ▓▓▓▓▄▓▓▓▓ ▐▓▓    ▓▓▓ ▓▓▓  ▓▓▓▄    ▓▓▓▓`
            '▀▓▓▓▄      ^▓▓▓  ▓▓▓       └▀▀▀▀ ▀▀ ^▀▀    `▀▀ `▀▀   '▀▀    ▐▓▓▌
               ▀▀▀▀▓▄▄▄   ▓▓▓▓▓▓,                                      ▓▓▓▓▀
                   `▀█▓▓▓▓▓▓▓▓▓▌
                        ¬`▀▀▀█▓


 Version information:
  ml-agents: 0.15.0,
  ml-agents-envs: 0.15.0,
  Communicator API: 0.15.0,
  TensorFlow: 2.0.1
WARNING:tensorflow:From c:\users\korfanty\unity\python3.7\lib\site-packages\tensorflow_core\python\compat\v2_compat.py:65: disable_resource_variables (from tensorflow.python.ops.variable_scope) is deprecated and will be removed in a future version.
Instructions for updating:
non-resource variables are not supported in the long term
Process Process-1:
Traceback (most recent call last):
  File "c:\users\korfanty\unity\python3.7\lib\multiprocessing\process.py", line 297, in _bootstrap
    self.run()
  File "c:\users\korfanty\unity\python3.7\lib\multiprocessing\process.py", line 99, in run
    self._target(*self._args, **self._kwargs)
  File "c:\users\korfanty\unity\python3.7\lib\site-packages\mlagents\trainers\subprocess_env_manager.py", line 97, in worker
    worker_id, [shared_float_properties, engine_configuration_channel]
  File "c:\users\korfanty\unity\python3.7\lib\site-packages\mlagents\trainers\learn.py", line 443, in create_unity_environment
    side_channels=side_channels,
  File "c:\users\korfanty\unity\python3.7\lib\site-packages\mlagents_envs\environment.py", line 142, in __init__
    aca_output = self.send_academy_parameters(rl_init_parameters_in)
  File "c:\users\korfanty\unity\python3.7\lib\site-packages\mlagents_envs\environment.py", line 549, in send_academy_parameters
    return self.communicator.initialize(inputs)
  File "c:\users\korfanty\unity\python3.7\lib\site-packages\mlagents_envs\rpc_communicator.py", line 98, in initialize
    self.poll_for_timeout()
  File "c:\users\korfanty\unity\python3.7\lib\site-packages\mlagents_envs\rpc_communicator.py", line 91, in poll_for_timeout
    "The Unity environment took too long to respond. Make sure that :\n"
mlagents_envs.exception.UnityTimeOutException: The Unity environment took too long to respond. Make sure that :
         The environment does not need user interaction to launch
         The Agents are linked to the appropriate Brains
         The environment and the Python interface have compatible versions.
2020-03-25 01:19:25 INFO [trainer_controller.py:104] Learning was interrupted. Please wait while the graph is generated.
2020-03-25 01:19:25 INFO [trainer_controller.py:100] Saved Model
```

**Screenshots**
Probably N/A

**Environment (please complete the following information):**
- Windows 10
- ML-Agents v.0.15.0
- _TensorFlow version_: 2.0.1
- _Environment_: Custom, but also tried with 3DBall with no avail.

**Comment by Nomadjackalope** [comment]
I'm having this issue but I have only installed ml-agents via Pypi in the instructions. I haven't tried the ones included in the repo.

**Comment by surfnerd** [comment]
Hi @RaStall, 
I'll try to reproduce your error.

**Comment by Nomadjackalope** [comment]
I tried just pressing play in the Unity editor when I was waiting for the timeout and it began training.

**Comment by RaStall** [comment]
Shouldn't mlagents display a message in the console with the prompt to Press play?

**Comment by surfnerd** [comment]
It used to, maybe it was removed.  Did you get it to work?

**Comment by surfnerd** [comment]
I see the print out is still there 
https://github.com/Unity-Technologies/ml-agents/blob/7507a5d3f5515ae7877eb3a3e9abaa2e2a270930/ml-agents-envs/mlagents_envs/environment.py#L132

If you press play in the editor after you launch mlagents-learn, you should see it work.

**Comment by RaStall** [comment]
Yeah, I was able to get it running by pressing play after launching mlagents-learn, but it's really confusing without the print, as most tutorials state that I should wait for "Start training by pressing the Play button in the Unity Editor." print in the console.

**Comment by surfnerd** [comment]
Thanks for the feedback.  This is being looked into today.  We’ll update you when it’s fixed.

**Comment by surfnerd** [comment]
Hey @RaStall,
This was fixed in #3703 and was put into the [0.15.0 release](https://github.com/Unity-Technologies/ml-agents/releases/tag/0.15.1).  I will close this for now.  Please re-open this issue if the problem persists.
